portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -46,14 +49,25 @@
    @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
     // Get the input from the form.
+    
     String text = request.getParameter("text-input");
     Entity taskEntity = new Entity("Task");
     taskEntity.setProperty("text", text);
- DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
- datastore.put(taskEntity);
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    datastore.put(taskEntity);
+ Query query = new Query("Task");
+    PreparedQuery results = datastore.prepare(query);
+   List<String> tasks = new ArrayList<>();
+ for (Entity entity : results.asIterable()) {
+      String comment = (String) entity.getProperty("text");
+      String task = new String(comment);
+      tasks.add(task);
+    }
+    Gson gson = new Gson();
     // Respond with the result.
     response.setContentType("text/html;");
-    response.getWriter().println(text);
+    response.getWriter().println(gson.toJson(tasks));
```

Comment:
If the post comes from html form (as opposed to javascript) then whatever response the servlet sends back will be rendered by browser. In this case browser will show contents of json. This is probably not what was intended. 

Typically handler for post requests redirects browser to a page that fetches and renders updated data (in our examples index.html). One case where sending data in response to post request makes sense is when post was initiated from client javascript, which expects some data back.

Reply:
so I should remove the json conversion?

Reply:
You can merge this PR as is, I already approved it. I left my comment to outline the potential issue, something for you to keep in mind.